src/script/sign.cpp

```diff
@@ -291,6 +279,15 @@ static bool SignMuSig2(const BaseSignatureCreator& creator, SignatureData& sigda
                 if (leaf_hash) it->second.first.insert(*leaf_hash);
             }
         }
+    }
+
+    // Compute sighash
```

Comment:
Maybe this could come before the derivation-path-filling loop above. That way the early exit condition on missing sighash comes earlier, and the loop does not have to be broken into two.